MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java

```diff
@@ -4786,22 +4787,75 @@ private void updateOptions() {
+        options.setMinimumHitsForVees((Integer) spnMinimumHitsForVees.getModel().getValue());
+        options.setUseRandomHitsForVees(useRandomHitsForVees.isSelected());
+        options.setTougherHealing(useTougherHealing.isSelected());
+        options.setUseTransfers(chkUseTransfers.isSelected());
+        options.setUseTimeInService(chkUseTimeInService.isSelected());
+        options.setShowOriginFaction(chkShowOriginFaction.isSelected());
+        options.setRandomizeOrigin(chkRandomizeOrigin.isSelected());
+        options.setRandomizeDependentOrigin(chkRandomizeDependentsOrigin.isSelected());
+        options.setOriginSearchRadius((Integer)spnOriginSearchRadius.getModel().getValue());
+        //Family
+        options.setMinimumMarriageAge((Integer) spnMinimumMarriageAge.getModel().getValue());
+        options.setCheckMutualAncestorsDepth((Integer) spnCheckMutualAncestorsDepth.getModel().getValue());
+        options.setUseRandomMarriages(chkUseRandomMarriages.isSelected());
+        options.setChanceRandomMarriages((Double) spnChanceRandomMarriages.getModel().getValue());
+        options.setMarriageAgeRange((Integer) spnMarriageAgeRange.getModel().getValue());
+        for (int i = 0; i < Person.NUM_SURNAME; i++) {
+            int val = (int) Math.round(((Double) spnRandomMarriageSurnameWeights[i].getModel().getValue()) * 10);
+            options.setRandomMarriageSurnameWeight(i, val);
         }
+        options.setUseRandomSameSexMarriages(chkUseRandomSameSexMarriages.isSelected());
+        options.setChanceRandomSameSexMarriages((Double) spnChanceRandomSameSexMarriages.getModel().getValue());
+        options.setUseUnofficialProcreation(chkUseUnofficialProcreation.isSelected());
+        options.setChanceProcreation((Double) spnChanceProcreation.getModel().getValue());
+        options.setUseUnofficialProcreationNoRelationship(chkUseUnofficialProcreationNoRelationship.isSelected());
+        options.setChanceProcreationNoRelationship((Double) spnChanceProcreationNoRelationship.getModel().getValue());
+        options.setDisplayTrueDueDate(chkDisplayTrueDueDate.isSelected());
+        options.setLogConception(chkLogConception.isSelected());
+        options.setBabySurnameStyle(comboBabySurnameStyle.getSelectedIndex());
+        options.setUseParentage(chkUseParentage.isSelected());
+        options.setDisplayFamilyLevel(comboDisplayFamilyLevel.getSelectedIndex());
+        options.setUseRandomDeaths(chkUseRandomDeaths.isSelected());
+        options.setKeepMarriedNameUponSpouseDeath(chkKeepMarriedNameUponSpouseDeath.isSelected());
+        //Salary
         options.setSalaryCommissionMultiplier((Double) spnSalaryCommission.getModel().getValue());
         options.setSalaryEnlistedMultiplier((Double) spnSalaryEnlisted.getModel().getValue());
         options.setSalaryAntiMekMultiplier((Double) spnSalaryAntiMek.getModel().getValue());
-        //end salary
+        for (int i = 0; i < 5; i++) {
```

Comment:
Should this instead use `spnSalaryXp.length()`?

Reply:
Yes, it should. Fixed.